src/units/filter.cpp

```diff
@@ -386,6 +408,42 @@ void unit_filter_compound::fill(vconfig cfg)
 			}
 		);
 
+		create_attribute(literal["ability_id_active"],
+			[](const config::attribute_value& c) { return utils::split(c.str()); },
+			[](const std::vector<std::string>& abilities, const unit_filter_args& args)
+			{
+				assert(display::get_singleton());
+				const unit_map& units = display::get_singleton()->get_units();
+				for(const std::string& ability : abilities) {
+					std::vector<ability_match> ability_id_matches_self;
+					get_ability_children_id(ability_id_matches_self, args.u.abilities(), ability);
+					for(const ability_match& entry : ability_id_matches_self) {
+						if (args.u.get_self_ability_bool(*entry.cfg, entry.tag_name, args.loc)) {
+							return true;
+						}
+					}
+
+					const auto adjacent = get_adjacent_tiles(args.loc);
+					for(unsigned i = 0; i < adjacent.size(); ++i) {
+						const unit_map::const_iterator it = units.find(adjacent[i]);
+						if (it == units.end() || it->incapacitated())
+							continue;
+						if ( &*it == (args.u.shared_from_this()).get() )
```

Comment:
Please remove the extra spaces, so it's `if (&*it == (args.u.shared_from_this()).get())` to keep this consistent.

Reply:
It done